### ISSUE TITLE
UTC是否支持多分类、多标签、层次分类、层次多标签 的多种还是一种？

### ISSUE THREAD

**Issue by zbmain** [PaddlePaddle/PaddleNLP#4973]
### 请提出你的问题

您好，在使用applications/text_classification/*/few-shot时，有mutil_class、mutil_label、hierarchical三种类目结构方案。
最近看到文档更新few-shot推荐使用UTC。但是看UTC的文档，并没有看到如何能支持不同类目结构，麻烦解疑一下：UTC可以支持的类目结构和方案？

**Comment by LemonNoel** [comment]
三种分类都可以在 UTC 框架下实现，其中 multi-class 需要修改评估代码。

1. multi-label: UTC 默认是 multi-label 形式
```
# 数据示例
{"text_a": "多标签分类示例", "choices": ["体育", "时政", "娱乐", "电影"], "labels": [2, 3], "text_b": "", "question": ""}
```
2. hierachical: applications/text_classification/*/few-shot 目录下 hierachical 的代码实现和 multi-label 是相同的，区别是数据层面将多级标签用 `##` 分隔符拼接了起来，形式上仍是 multi-label。在 UTC 中也可以使用这种实现，将多级标签直接拼起来，如果只需要分类到细分层级，直接取细分层级效果可能更好。
```
# 数据示例
{"text_a": "多层次分类示例", "choices": ["环境 资质优",  "环境 资质差", "口味 口感好", "口味 口感差"], "labels": [0, 1],  "text_b": "", "question": ""}
```
3. multi-class: 与上述两种分类的区别是 multi-class 最终分类结果有且只有一个标签。
```
# 数据示例
{"text_a": "多分类示例", "choices": ["体育", "时政", "娱乐"], "labels": 0, "text_b": "", "question": ""}
```
- 修改代码部分：原有代码中compute_metrics 函数的 sigmoid 实现 [run_train.py](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/applications/zero_shot_text_classification/run_train.py#L99-L106) 和 [run_eval.py](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/applications/zero_shot_text_classification/run_eval.py#L78-L85) 均改为 softmax
```
def compute_metrics(eval_preds):
        labels = paddle.to_tensor(eval_preds.label_ids, dtype="int64")
        preds = paddle.to_tensor(eval_preds.predictions)
        preds = paddle.nn.functional.softmax(preds, axis=-1)
        preds = preds[labels != -100]
        labels = paddle.argmax(labels, axis=-1)

        metric = Accuracy()
        correct = metric.compute(preds, labels)
        metric.update(correct)
        acc = metric.accumulate()
        return {"accuracy": acc}
```

**Comment by zbmain** [comment]
感谢，很清晰的解惑。
